airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -617,15 +619,19 @@ def on_kill(self):
             self._submit_sp.kill()
 
             if self._yarn_application_id:
-                self.log.info('Killing application %s on YARN', self._yarn_application_id)
-
                 kill_cmd = "yarn application -kill {}" \
                     .format(self._yarn_application_id).split()
+                if self._keytab is not None and self._principal is not None:
+                    renew_from_kt(self._principal, self._keytab, False)
+                    krb5_env = "env KRB5CCNAME={}" \
+                        .format(airflow_conf.get('kerberos', 'ccache')).split()
+                    kill_cmd = krb5_env.extend(kill_cmd)
+
                 yarn_kill = subprocess.Popen(kill_cmd,
                                              stdout=subprocess.PIPE,
                                              stderr=subprocess.PIPE)
```

Comment:
Popen can take then environment dictionary directly:

```suggestion
                env=None
                if self._keytab is not None and self._principal is not None:
                    renew_from_kt(self._principal, self._keytab, False)

                    kenv = os.environ.copy()
                    env["KRB5CCNAME"] = conf.get('kerberos', 'ccache')

                yarn_kill = subprocess.Popen(kill_cmd,
                                             env=env,
                                             stdout=subprocess.PIPE,
                                             stderr=subprocess.PIPE)
```